### PR TITLE
Adds linux support. Replaces shared:glfw, and shared:opengl with vendor:glfw, and vendor:opengl.

### PR DESCRIPTION
**Pull request by NoahR02**

### CODE REVIEW

File: impl/opengl/opengl.odin [hunk 1]
```diff
@@ -211,10 +211,19 @@ restore_opengl_state :: proc(state: OpenGL_Backup_State) {
                          u32(state.last_blend_src_alpha), 
                          u32(state.last_blend_dst_alpha));
 
-    if state.last_enabled_blend       do gl.Enable(gl.BLEND)        else do gl.Disable(gl.BLEND);
-    if state.last_enable_cull_face    do gl.Enable(gl.CULL_FACE)    else do gl.Disable(gl.CULL_FACE);
-    if state.last_enable_depth_test   do gl.Enable(gl.DEPTH_TEST)   else do gl.Disable(gl.DEPTH_TEST);
-    if state.last_enable_scissor_test do gl.Enable(gl.SCISSOR_TEST) else do gl.Disable(gl.SCISSOR_TEST);
+    if state.last_enabled_blend {
```
Comment (ThisDevDane) [inline impl/opengl/opengl.odin]: Why the format change here?

Comment (NoahR02) [reply impl/opengl/opengl.odin]: Fixed.

File: impl/opengl/opengl.odin [hunk 2]
```diff
@@ -249,7 +249,7 @@ setup_imgui_shaders :: proc() -> u32 {
     
     ok: i32;
     gl.GetProgramiv(program_h, gl.LINK_STATUS, &ok);
-    if ok != gl.TRUE {
+    if ok != 1 {
```
Comment (ThisDevDane) [inline impl/opengl/opengl.odin]: Why not `gl.TRUE`? Other places like line 195 uses it?

Comment (NoahR02) [reply impl/opengl/opengl.odin]: In shared:opengl, TRUE was 1. In vendor:opengl, TRUE is true. I can cast ok to a bool and compare it that way.

**Comment by sumofat** [comment]
Hey I think it would be great if you could add your Linux check and changes to the PR I did for osx and win

https://github.com/ThisDrunkDane/odin-imgui/pull/9

**Pull request closed without merging**